### PR TITLE
Use readfile() for serving protected uploads

### PR DESCRIPTION
**Pull request by leedxw**
I suspect that one performance problem we have stems from loading binary files into memory, and then using `echo()` to send them to the client.

This switches off output buffering and uses `readfile()`.

- [ ] Version number has been bumped

### CODE REVIEW

**Review by snim2 — CHANGES_REQUESTED** [review]
So, I've tested this locally, and it still serves assets, so I think the change is fine. However, we need to update the Changelog before merging this.

**Comment by leedxw** [comment]
You need to update the changelog, version string, git tagged version.

**Comment by RobjS** [comment]
Do we have a recommended local config for testing this plugin on files in `uploads/`? By default, requests for files in that directory aren't routed via WordPress auth, so this plugin code doesn't run.

**Comment by leedxw** [comment]
Is this for saluki testing?

`ENABLE_INTRANET_SETTINGS=true`

will force uploads to be delivered via `index.php`

**Comment by RobjS** [comment]
No, this is for local testing using our `thedxw/wpc-wordpress` image

**Comment by snim2** [comment]
We've merged other things since this was raised so the version number needs to be checked

**Review by RobjS — COMMENTED** [review]
I've tested this locally with Saluki and it's working, just needs version number updating.

File: readme.txt [hunk 1]
```diff
@@ -2,31 +2,37 @@
...
-    - IPs that can view the site without logging in
+    - IP addresses that can view the site without logging in
     - Where to redirect visitors who are not logged in and try to view restricted content
     - Whether to automatically restrict access to uploads by default
     - A max age for the cache-control header that will be served to any users who try to access restricted content when not logged in
 
 == Development ==
 
 https://github.com/dxw/dxw-members-only
+
```
Comment (snim2) [inline readme.txt:33]: @RobjS there are two README files in this plugin and this one is not displayed by default in GitHub. I'm inclined to delete it. wdyt?

Comment (RobjS) [reply readme.txt:33]: Yes, I think this is an artifact of when we published the plugin on the WordPress directory, but that's not the case in anymore, so happy to delete.

**Review by snim2 — APPROVED** [review]

File: readme.txt (hunk 1 above)
Comment (leedxw) [reply readme.txt]: If it's a WordPress plugin it should be using the Wordpress plugin conventions regardless of if it's published imo.

**Pull request merged**